adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImpl.java

```diff
@@ -81,45 +84,39 @@ public Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsStrin
     /**
      * This method converts a number of seconds elapsed since noon of say of service to a string formatted as HH:MM:SS.
      *
-     * @param elapsedDurationSinceNoon the number of seconds elapsed since noon of say of service to be converted as a
-     *                                 string formatted as HH:MM:SS.
+     * @param elapsedDurationSinceNoonInSeconds the number of seconds elapsed since noon of day of service to be
+     *                                          converted as a string formatted as HH:MM:SS.
      * @return the human readable string representation of the number of seconds elapsed since noon of day of service.
      * This string is formatted as follows: HH:MM:SS.
      */
-    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoon) {
-        if (elapsedDurationSinceNoon != null) {
-            // Determine number of hours elapsed
-            int hourValue = (int) (TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon) + 12);
-            int minuteValue;
-            int secondValue;
-            // elapsedDurationSinceNoon > 0 for times after noon, e.g 14:00PM is represented a +2*3600s=+7200s
-            // elapsedDurationSinceNoon < 0 for times before noon, e.g 11:00AM is represented as -1*3600s=-3600s
-            if (elapsedDurationSinceNoon > 0) {
-                // Determine number of minutes and seconds elapsed
-                minuteValue = (int) (TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon) -
-                        TimeUnit.HOURS.toMinutes(TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon)));
-                secondValue = (int) (TimeUnit.SECONDS.toSeconds(elapsedDurationSinceNoon) -
-                        TimeUnit.MINUTES.toSeconds(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon)));
+    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoonInSeconds) throws
+            IllegalArgumentException {
+        if (elapsedDurationSinceNoonInSeconds == null) {
+            throw new IllegalArgumentException("elapsedDurationSinceNoonInSeconds cannot be null");
+        } else {
```

Comment:
For early returns like this, you typically don't wrap the rest of the method in an `else`.  It would just be:

```java
if (elapsedDurationSinceNoonInSeconds == null) {
    throw new IllegalArgumentException("elapsedDurationSinceNoonInSeconds cannot be null");
}
```

...followed by the rest of the code. This way the rest of the code is simpler to trace, doesn't indent as much, and indicates that you can't execute the remainder of the body if `null`.

Reply:
Oh okay! I understand, thanks.